### PR TITLE
Signing apps

### LINKED ISSUE #294: Publish helpwave tasks
### Describe the feature

Add a necessary changes to make `helpwave tasks` available on Google Play Store and Apple App Store.

### Why is this feature needed?

We want `helpwave tasks` to be available on the Google Play Store and Apple App Store.

### PR DESCRIPTION
**Pull request by use-to**
## Which issues does this pull request close?
closes #294 

Prepared tasks and impulse for android release builds.

- [x] Google Play Store

### CODE REVIEW

**Comment by MaxSchaefer** [comment]
@use-to can we close this pr? The To-dos of this pull request should be tracked via an Issue instead. Outstanding requested changes from Google as well.

**Comment by use-to** [comment]
> @use-to can we close this pr? The To-dos of this pull request should be tracked via an Issue instead. Outstanding requested changes from Google as well.

Okay, I'll wait with the remaining issues until we got the App Store access.
r4r

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -109,7 +109,7 @@ jobs:
       - name: package tasks ipa
         working-directory: ./apps/tasks/
         run: |
-          flutter build ios --no-codesign
+          flutter build ios --no-codesign #--debug
```
Comment (MaxSchaefer) [inline .github/workflows/ci.yaml:112]: `#--debug`?

File: apps/impulse/android/app/build.gradle [hunk 2]
```diff
@@ -22,6 +22,12 @@ if (flutterVersionName == null) {
     flutterVersionName = '1.0'
 }
 
+ def keystoreProperties = new Properties()
+ def keystorePropertiesFile = rootProject.file('key.properties')
```
Comment (MaxSchaefer) [inline apps/impulse/android/app/build.gradle:26]: Should this file be available in the project? Does this file contain our singing key? Please add a comment that this fails on dev machines and is an intended behavior.

File: melos.yaml [hunk 3]
```diff
@@ -12,13 +12,19 @@ scripts:
     run: melos exec --file-exists="l10n.yaml" -- "flutter gen-l10n && melos bootstrap"
 
   build-tasks-apk:
```
Comment (MaxSchaefer) [inline melos.yaml:14]:
```suggestion
  build-tasks-apk-debug:
```
The same goes for the other scripts as well.

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: apps/tasks/pubspec.yaml [hunk 4]
```diff
@@ -15,7 +15,7 @@ publish_to: 'none'
 # https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
 # In Windows, build-name is used as the major, minor, and patch parts
 # of the product and file versions while build-number is used as the build suffix.
-version: 1.0.0+1
+version: 1.0.0+2
```
Comment (MaxSchaefer) [inline apps/tasks/pubspec.yaml:18]: It is fine, I guess. This gets override by flutter build?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (use-to) [reply .github/workflows/ci.yaml:112]: We need that later on... If we build an ios app for release, we need to specify the debug flag (see android)

File: apps/impulse/android/app/build.gradle (hunk 2 above)
Comment (use-to) [reply apps/impulse/android/app/build.gradle:26]: It's not intended to check in our key. So this is required.
I don't really see where this results in a failure on dev...
While developing, you're not building the app with the release target.

File: apps/tasks/pubspec.yaml (hunk 4 above)
Comment (use-to) [reply apps/tasks/pubspec.yaml:18]: This is version increment is required as we already published two versions on the play store.
Already mentioned that, we need to get a proper versioning for all of our projects working asap.

File: melos.yaml (hunk 3 above)
Comment (use-to) [reply melos.yaml:14]: No, I would like to define the default build configuration as `debug` as it's not possible to build a release on your own without using a proper key. If we need a release build command in `melos` later on, we should name it `build-tasks-apk-release`. But very likely we don't need a separate melos command for that.

Comment (MaxSchaefer) [reply melos.yaml:14]: > No, I would like to define the default build configuration as debug as it's not possible to build a release on your own without using a proper key. 

@use-to got it. Good idea. But what keeps us from being more specific in the name of the configuration? We are doing the same in the CI as well.

Comment (use-to) [reply melos.yaml:14]: No, it's different to the CI configuration, as the CI will also build a release version later on. So we need to distinguish between debug and release. Why should I add space for a configuration a developer does not need or intent to use. Can we just leave it as it is and move this discussion? I don't see any more valid arguments on both sides.

Comment (MaxSchaefer) [reply melos.yaml:14]: _moved_

**Review by MaxSchaefer — APPROVED** [review]

**Pull request merged**